homeassistant/components/homee/entity.py

```diff
@@ -0,0 +1,147 @@
+"""Base Entities for Homee integration."""
+
+from typing import Any
+
+from pyHomee.const import AttributeType, NodeProfile
+from pyHomee.model import HomeeAttribute, HomeeNode
+
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import Entity
+
+from . import HomeeConfigEntry
+from .const import ATTR_HOMEE_DATA, CONF_ADD_HOMEE_DATA, DOMAIN
+from .helpers import get_name_for_enum
+
+
+class HomeeNodeEntity(Entity):
+    """Representation of a Node in Homee."""
+
+    _unrecorded_attributes = frozenset({ATTR_HOMEE_DATA})
+
+    def __init__(self, node: HomeeNode, entry: HomeeConfigEntry) -> None:
+        """Initialize the wrapper using a HomeeNode and target entity."""
+        self._node = node
+        self._clear_node_listener = None
+        self._attr_unique_id = node.id
+        self._entry = entry
+
+        self._homee_data = {
+            "id": node.id,
+            "name": node.name,
+            "profile": node.profile,
+            "attributes": [{"id": a.id, "type": a.type} for a in node.attributes],
+        }
+        self._attr_device_info = dr.DeviceInfo(
+            identifiers={(DOMAIN, self._node.id)},
+            name=self._node.name,
+            manufacturer="unknown",
+            model=get_name_for_enum(NodeProfile, self._homee_data["profile"]),
+            sw_version=self._get_software_version(),
+            via_device=(DOMAIN, self._entry.runtime_data.settings.uid),
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Add the homee binary sensor device to home assistant."""
+        self.async_on_remove(self._node.add_on_changed_listener(self._on_node_updated))
+
+    @property
+    def available(self) -> bool:
+        """Return the availability of the underlying node."""
+        return self._node.state <= 1
+
+    @property
+    def should_poll(self) -> bool:
+        """Return if the entity should poll."""
+        return False
+
+    @property
+    def raw_data(self):
+        """Return the raw data of the node."""
+        return self._node.raw_data
+
+    @property
+    def extra_state_attributes(self) -> dict[str, dict[str, Any]] | None:
+        """Return entity specific state attributes."""
+        data = {}
+
+        if self._entry.options.get(CONF_ADD_HOMEE_DATA, False):
+            data[ATTR_HOMEE_DATA] = self._homee_data
+
+        return data if data else None
+
+    async def async_update(self):
+        """Fetch new state data for this node."""
+        homee = self._entry.runtime_data
+        await homee.update_node(self._node.id)
```

Comment:
How does the data here exactly work? Is this cached in the library? What if I have a device that has 5 entites and I use the `homeassistant.update_entity` service (which triggers an update), would that trigger 5 requests to the API?

Reply:
This function will make a call to the Api to request data for that node [1]. If the Node has 5 Attributes, this is still only one call which returns info on the whole node including all attributes.
But you are right, there is also a function tu just update a single attribute (right below it in the code), which would be more fitting for an entity. 
It gets a little unsharp in case of entities that use multiple attributes. Cover does so for up/down movements, position, slat movements, ...
So maybe it's a good idea to keep it here and overwrite the function for entities that have a single attribute.

[1]: https://github.com/Taraman17/pyHomee/blob/d1c3e397a61abadf6fa5d6e9f37923460b8db833/pyHomee/__init__.py#L501

Reply:
I mean more like, if one API call has data for multiple entities, maybe we should introduce a coordinator to handle that data. A coordinator is a centralized way to manage state for multiple entities.

But what does this function exactly do if its push based?

Reply:
It gives the user the chance tu manually request an update, if the push is not often enough.
Like I had one user who didn't get the position of his garage door quick enough, so it is the chance of manually update it with an action.

Reply:
I mean more like, if one API call has data for multiple entities, maybe we should introduce a coordinator to handle that data. A coordinator is a centralized way to manage state for multiple entities.

But what does this function exactly do if its push based?

Reply:
This is indeed a poll. It sends an API command to get the current state of the node.

Reply:
But what makes this integration local push?

Reply:
That everything else besides this possibility to manually request an update is push.
Does this one exemption make it another type?

Reply:
Okay so this call is just requesting another push to happen?

Reply:
Yes, thats the perfect way to put it. ;-)

Reply:
Do we just keep it then?

Reply:
Check discord